compiler/rustc_abi/src/layout/ty.rs

```diff
@@ -288,7 +295,7 @@ impl<'a, Ty> TyAndLayout<'a, Ty> {
                 // More than one non-1-ZST field.
                 return None;
             }
-            found = Some((field_idx, field));
+            found = Some((FieldIdx::from_usize(field_idx), field));
```

Comment:
it may be worth adding an `iter` and an `iter_enumerated` method to `FieldsShape`